http/custom_roundtripper.go

```diff
@@ -0,0 +1,54 @@
+package http
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	awsAuth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+)
+
+type AwsSignerRoundTripper struct {
+	signer       *awsAuth.Signer
+	roundTripper http.RoundTripper
+}
+
+func NewAWSSignerRoundTripper(awsFilename, awsProfile, awsRegion, awsService string, customTransport http.RoundTripper) *AwsSignerRoundTripper {
+	var roundTripper http.RoundTripper
+	if awsRegion == "" || awsService == "" {
+		log.Fatal("aws region and service should be valid options")
```

Comment:
I've been thinking and I think it's much better to return nil and an error here. I don't think an external package should be deciding how my program exits, even moreso considering os.Exit(1) will skip any deferred functions higher up in the stack.

Reply:
Hmm.. changed it now in update 2. Not convinced though on the wrapping of errors as i think errors should not be wrapped unnecessarily.